spec/database_cleaner/active_record/truncation_spec.rb

```diff
@@ -1,113 +1,96 @@
+        helper.setup
+        example.run
+        helper.teardown
       end
-    end
-  end
-end
-
-module DatabaseCleaner
-  module ActiveRecord
 
-    RSpec.describe Truncation do
-      let(:connection) { double('connection') }
+      let(:connection) { helper.connection }
 
-      before(:each) do
+      before do
         allow(connection).to receive(:disable_referential_integrity).and_yield
         allow(connection).to receive(:database_cleaner_view_cache).and_return([])
-        allow(::ActiveRecord::Base).to receive(:connection).and_return(connection)
       end
 
       describe '#clean' do
-        it "should truncate all tables except for schema_migrations" do
-          allow(connection).to receive(:database_cleaner_table_cache).and_return(%w[schema_migrations widgets dogs])
-
-          expect(connection).to receive(:truncate_tables).with(['widgets', 'dogs'])
-          Truncation.new.clean
-        end
-
-        it "should use ActiveRecord's SchemaMigration.table_name" do
-          allow(connection).to receive(:database_cleaner_table_cache).and_return(%w[pre_schema_migrations_suf widgets dogs])
-          allow(::ActiveRecord::Base).to receive(:table_name_prefix).and_return('pre_')
-          allow(::ActiveRecord::Base).to receive(:table_name_suffix).and_return('_suf')
-
-          expect(connection).to receive(:truncate_tables).with(['widgets', 'dogs'])
-
-          Truncation.new.clean
-        end
-
-        it "should only truncate the tables specified in the :only option when provided" do
-          allow(connection).to receive(:database_cleaner_table_cache).and_return(%w[schema_migrations widgets dogs])
-
-          expect(connection).to receive(:truncate_tables).with(['widgets'])
-
-          Truncation.new(:only => ['widgets']).clean
-        end
-
-        it "should not truncate the tables specified in the :except option" do
-          allow(connection).to receive(:database_cleaner_table_cache).and_return(%w[schema_migrations widgets dogs])
+        context "with records" do
+          before do
+            2.times { User.create! }
+            2.times { Agent.create! }
+          end
 
-          expect(connection).to receive(:truncate_tables).with(['dogs'])
+          it "should truncate all tables" do
+            expect { subject.clean }
+              .to change { [User.count, Agent.count] }
+              .from([2,2])
+              .to([0,0])
+          end
 
-          Truncation.new(:except => ['widgets']).clean
-        end
+          it "should reset AUTO_INCREMENT index of table" do
+            subject.clean
+            expect(User.create.id).to eq 1
+          end
 
-        it "should raise an error when :only and :except options are used" do
-          expect {
-            Truncation.new(:except => ['widgets'], :only => ['widgets'])
-          }.to raise_error(ArgumentError)
-        end
+          xit "should not reset AUTO_INCREMENT index of table if :reset_ids is false" do
```

Comment:
@botandrose Why was this one failing? Maybe you can create an issue for this so that we don't forget about it.

Reply:
Good call. This was the result of extending an existing test to cover more dbs. After doing so, it turns out that this doesn't work for some reason on some dbs. New information! I will create an issue

Reply:
Whoops, I was mistaken! I got this mixed up with another spec that had the situation I described! In any event, I have created an issue for this: #559